game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -99,35 +100,28 @@ static RulesAttachment get(
       final Collection<GamePlayer> playersToSearch,
       final boolean allowNull) {
     RulesAttachment ra = (RulesAttachment) player.getAttachment(nameOfAttachment);
-    if (ra == null) {
-      if (playersToSearch == null) {
-        if (!allowNull) {
-          throw new IllegalStateException(
-              "Rules & Conditions: No rule attachment for:"
-                  + player.getName()
-                  + " with name: "
-                  + nameOfAttachment);
-        }
-      } else {
-        for (final GamePlayer otherPlayer : playersToSearch) {
-          if (otherPlayer.equals(player)) {
-            continue;
-          }
-          ra = (RulesAttachment) otherPlayer.getAttachment(nameOfAttachment);
-          if (ra != null) {
-            return ra;
-          }
+    if (ra != null) {
+      return ra;
+    }
+    if (playersToSearch != null) {
```

Comment:
Done.

Reply:
Nice, I hope the other usage is guaranteed to be non-null otherwise that's going to be a problem (for obvious reasons the other usage you mentioned is not in the diff so I can't tell)

Reply:
Feel free to merge if the other usage is safe 👍

Reply:
Yep, the other usage was non-null.